hyperspy/learn/mva.py

```diff
@@ -1325,22 +1377,672 @@ def _estimate_elbow_position(self, curve_values):
+            scaling = scale_method
+            See scaling methods in scikit-learn preprocessing for further
+            details.
+        scaling_kwargs :
+            Additional parameters passed to the cluster scaling algorithm.
+            See sklearn.preprocessing scaling methods for further details
+        number_pca_components: int , default None
+            If you are clustering using the decomposition results
+            (use_decomposition_results = True) you can define how many PCA
+            components to use.   if set to None the method uses the
+            estimate of significant components found in the decomposition step
+            using the elbow method
+            (learningresults.number_significant_components)
+        navigation_mask : boolean numpy array
+            The navigation locations marked as True are not used in the
+            clustering.
+        signal_mask : boolean numpy array
+            The signal locations marked as True are not used in the
+            clustering.  Note that if use_decomposition_results=True the
+            signal_mask is ignored. The number of PCA components is used to
+            set the number of components to use.
+
+        Returns
+        ----------
+        scaled_data : numpy array - unfoled array of shape (number_of_samples,
+        no_of_features) scaled according to the selected algorithm
+
+        """
+
+        if import_sklearn.sklearn_installed is False:
+            raise ImportError(
+                'sklearn is not installed. Nothing done')
+        if scaling == "norm":
+            algorithm = import_sklearn.sklearn.preprocessing.Normalizer
+        elif scaling == "standard":
+            algorithm = import_sklearn.sklearn.preprocessing.StandardScaler
+        elif scaling == "minmax":
+            algorithm = import_sklearn.sklearn.preprocessing.MinMaxScaler
+        else:
+            algorithm = scaling
+            if scaling is not None:
+                if inspect.isclass(scaling) \
+                    and not issubclass(algorithm,
+                                import_sklearn.sklearn.base.TransformerMixin):
+                        raise ValueError("The class provided to the scaling"
+                            "parameter must be a scikit-learn"
+                            "preprocessing class.")
+
+        if self.axes_manager.navigation_size < 2:
+            raise AttributeError("It is not possible to cluster a dataset "
+                                 "with navigation_size < 2")
+
+        self._unfolded4clustering = self.unfold()
+        # Deal with masks
+        navigation_mask = _mask_for_clustering(navigation_mask)
+        if use_decomposition_results:
+            dc = self.learning_results.loadings.copy()
+            data = dc[:,slice(0,number_pca_components,1)]
+            signal_mask =  _mask_for_clustering(None)
+        else:
+            data = self.data
```

Comment:
if clustering the raw data then if depended if navigation axes was always 0 index in the array so the line below (think its 1753 now with the refactor) was just a check on which way around it is rather than just assuming .T is correct